src/plugins/mmapstorage/mmapstorage.c

```diff
@@ -48,40 +49,39 @@ static MmapMetaData magicMmapMetaData;
 /* -- File handling --------------------------------------------------------------------------------------------------------------------- */
 
 /**
- * @brief Wrapper for fopen().
+ * @brief Wrapper for open().
  *
  * @param parentKey containing the filename
- * @param mode file open mode
+ * @param flags file access mode
  *
- * @return file pointer
+ * @return file descriptor
  */
-static FILE * openFile (Key * parentKey, const char * mode)
+static int openFile (Key * parentKey, int flags)
 {
-	FILE * fp;
+	int fd;
 	ELEKTRA_LOG_DEBUG ("opening file %s", keyString (parentKey));
 
-	if ((fp = fopen (keyString (parentKey), mode)) == 0)
+	if ((fd = open (keyString (parentKey), flags, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH)) == -1)
```

Comment:
Why do you give write permissions? You always unlink the file, so write permissions should be never needed.

Reply:
@markus2330 would you in general prefer minimal access (only user can read)?

I think the maximum should be user + group access. Preferrably even only S_IRUSR. If others have access, it might leak data that should not be readable by others.

Reply:
I tested it and the minimal access mode bits are `S_IRUSR | S_IWUSR`.

`mmapstorage` works with less, but it introduces problems with other parts of the tests. (1/3 of the tests fail on my machine)